lib/SPIRV/SPIRVToLLVMDbgTran.cpp

```diff
@@ -1412,6 +1410,13 @@ MDNode *SPIRVToLLVMDbgTran::transExpression(const SPIRVExtInst *DebugInst) {
   return getDIBuilder(DebugInst).createExpression(Addr);
 }
 
+DIType *
```

Comment:
Just curious. why do we need a separate function here? Can we just not return the unknown type inside the 'case DebugInfoNone' construct inside transDebugInstImpl?

Just a nit.

Thanks

Reply:
Because `transDebugInstImpl` handles every debug instruction, not only types. And we can't even specialize it for <DIType> for the reasons I mentioned in the description:
`This function doesn't replace all calls to transDebugInst<DIType> as there are cases, where we can generate null type, for example DWARF doesn't require it for DW_TAG_typedef, hence I'm not changing translation flow in this case. Additionally to this, while DWARF requires type attribute for DW_TAG_pointer_type, LLVM does not, hence I'm not changing translation flow in this case as well.`

Reply:
Ah. Got it. Thanks for pointing back to the description.